src/server/ProcessList.cpp

```diff
@@ -30,51 +27,27 @@ using namespace Microsoft::Console::Interactivity;
 [[nodiscard]] HRESULT ConsoleProcessList::AllocProcessData(const DWORD dwProcessId,
                                                            const DWORD dwThreadId,
                                                            const ULONG ulProcessGroupId,
-                                                           _In_opt_ ConsoleProcessHandle* const pParentProcessData,
                                                            _Outptr_opt_ ConsoleProcessHandle** const ppProcessData)
 {
-    FAIL_FAST_IF(!(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked()));
+    assert(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked());
 
     auto pProcessData = FindProcessInList(dwProcessId);
-    if (nullptr != pProcessData)
+    if (!pProcessData)
     {
-        // In the GenerateConsoleCtrlEvent it's OK for this process to already have a ProcessData object. However, the other case is someone
-        // connecting to our LPC port and they should only do that once, so we fail subsequent connection attempts.
-        if (nullptr == pParentProcessData)
-        {
-            return E_FAIL;
-            // TODO: MSFT: 9574803 - This fires all the time. Did it always do that?
-            //RETURN_HR(E_FAIL);

```

Comment:
TODO resolved: There's only two callers. `IoDispatchers::ConsoleHandleConnectionRequest` calls this method with a `nullptr` for `pParentProcessData`.

The other caller is `ApiDispatchers::ServerGenerateConsoleCtrlEvent`, which already has a `RETURN_HR_IF_NULL(E_INVALIDARG, pParentProcessData)`, preventing this method from ever being called in the first place.

Reply:
...unless of course `ConsoleHandleConnectionRequest`'s intention is to guard against accepting a connection twice? Is that something we need to guard against?